lib/model/CProbabilityAndInfluenceCalculator.cc

```diff
@@ -827,6 +859,42 @@ bool CProbabilityAndInfluenceCalculator::calculate(double& probability) const {
     return m_Probability.calculate(probability);
 }
 
+bool CProbabilityAndInfluenceCalculator::calculateExplainingProbabilities(
+    TStrDoubleUMap& explainingProbabilities) const {
+
+    double probability{0.0};
+    for (const auto& ep : m_ExplainingProbabilities) {
+        if (!ep.second.calculate(probability)) {
+            return false;
+        } else {
+            explainingProbabilities.emplace(ep.first, probability);
+        }
+    }
+
+    return true;
+}
+
+bool CProbabilityAndInfluenceCalculator::calculateMultiBucketImpact(double& multiBucketImpact) const {
```

Comment:
I think we should add some explanation for this function, i.e. something along the lines of "we choose a function s.t. the impact saturates when one "probability" < "other probability" / 1000 or one probability is close to one, i.e. one factor is not at all anomalous".